src/exec/mod.rs

```diff
@@ -201,21 +207,33 @@ impl ArgumentTemplate {
         }
     }
 
-    pub fn generate(&self, path: impl AsRef<Path>) -> OsString {
+    /// Generate an argument from this template. If path_separator is Some, then it will replace
+    /// the path separator in all placeholder tokens. Text arguments and tokens are not affected by
+    /// path separator substitution.
+    pub fn generate(&self, path: impl AsRef<Path>, path_separator: Option<&str>) -> OsString {
         use self::Token::*;
+        let path = path.as_ref();
 
         match *self {
             ArgumentTemplate::Tokens(ref tokens) => {
                 let mut s = OsString::new();
                 for token in tokens {
+                    // note: the s.push() and replace_separator() calls can't be pulled out of this
+                    // match or else we get errors about references to temporary values
```

Comment:
True, but I think that's to be expected. The alternative would be to copy or move the strings, but I think this is clearer.

If it's up to me, we can remove this comment.

Reply:
Yeah, I think that expected, but I fought with the borrow checker a bit here so I tend to document that. I can remove the comment though if you prefer.